packages/react-menu/src/MenuItem/renderMenuItem.tsx

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import { getSlots } from '@fluentui/react-utils';
+import { MenuItemState } from './MenuItem.types';
+import { menuItemShorthandProps } from './useMenuItem';
+
+/**
+ * Function that renders the final JSX of the component
+ * @param state Component state
+ */
+export const renderMenuItem = (state: MenuItemState) => {
+  const { slots, slotProps } = getSlots(state, menuItemShorthandProps);
+
+  return (
+    <slots.root {...slotProps.root}>
+      <slots.icon {...slotProps.icon} />
```

Comment:
https://github.com/microsoft/fluentui/blob/master/packages/react-menu/Spec.md#menu-items-with-icons

```html
<div role="menuitem" tabindex="0" class="">
  <!-- our slots.icon -->
  <span class="">
    <!-- an icon span -->
    <span role="presentation" aria-hidden="true" class="">
      <svg></svg>
    </span>
  </span>
  Item
</div>
```

This has a different output than in the spec. I am totally fine with it, but I should mention it 🔍

Reply:
Will bring it up in convergence touchpoints and try to get it in a RFC, can't be helped for this PR I'm afraid